src/main/java/cn/nukkit/command/defaults/BanCommand.java

```diff
@@ -12,17 +14,19 @@
  * author: MagicDroidX
  * Nukkit Project
  */
-public class BanCommand extends VanillaCommand {
-
-    public BanCommand(String name) {
-        super(name, "commands.ban.description", "/ban <player> [reason]");
-        this.setPermission("nukkit.command.ban.player");
-        this.commandParameters.clear();
-        this.commandParameters.add(
-                new CommandParameter[]{
-                        new CommandParameter("player", CommandParamType.TARGET, false),
-                        new CommandParameter("reason", CommandParamType.STRING, true)
-                });
+public class BanCommand extends Command {
+
+    public BanCommand() {
+        super("ban", CommandData.builder("ban")
+                .setDescription("commands.ban.description")
+                .setUsageMessage("/ban <player> [reason]")
+                .setPermissions("nukkit.command.ban.player")
+                .setParams(
+                        new CommandParameter[]{
```

Comment:
why declare the array explicitly? the method is varargs for just this reason

Reply:
It is a Varargs of Arrays, each set of Parameters needs to be an array.

Reply:
ah, i didn't notice that. whoops